modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/CopyEntriesMVCActionCommand.java

```diff
@@ -98,74 +110,110 @@ private void _checkDestinationGroup(
 		}
 	}
 
-	private void _copyFolder(
-			ActionRequest actionRequest, ActionResponse actionResponse)
-		throws IOException {
+	private List<String> _copyEntries(ActionRequest actionRequest)
+		throws Exception {
+
+		List<String> errorMessages = new ArrayList<>();
 
 		ThemeDisplay themeDisplay = (ThemeDisplay)actionRequest.getAttribute(
 			WebKeys.THEME_DISPLAY);
 
-		long sourceRepositoryId = ParamUtil.getLong(
-			actionRequest, "sourceRepositoryId");
-		long sourceFolderId = ParamUtil.getLong(
-			actionRequest, "sourceFolderId");
+		long destinationFolderId = ParamUtil.getLong(
+			actionRequest, "destinationParentFolderId");
 		long destinationRepositoryId = ParamUtil.getLong(
 			actionRequest, "destinationRepositoryId");
-		long destinationParentFolderId = ParamUtil.getLong(
-			actionRequest, "destinationParentFolderId");
-
-		try {
-			Group group = _groupLocalService.getGroup(destinationRepositoryId);
-
-			long[] groupIds =
-				_siteConnectedGroupGroupProvider.
-					getCurrentAndAncestorSiteAndDepotGroupIds(
-						group.getGroupId());
+		long sourceRepositoryId = ParamUtil.getLong(
+			actionRequest, "sourceRepositoryId");
 
-			_checkDestinationGroup(group, groupIds, sourceRepositoryId);
+		Group group = _groupLocalService.fetchGroup(destinationRepositoryId);
 
-			_dlAppService.copyFolder(
-				sourceRepositoryId, sourceFolderId, destinationRepositoryId,
-				destinationParentFolderId,
-				_getFileEntryTypeIds(group.getGroupId(), sourceFolderId),
-				groupIds,
-				ServiceContextFactory.getInstance(
-					DLFolder.class.getName(), actionRequest));
+		long[] groupIds =
+			_siteConnectedGroupGroupProvider.
+				getCurrentAndAncestorSiteAndDepotGroupIds(group.getGroupId());
+
+		Group sourceGroup = _groupLocalService.fetchGroup(sourceRepositoryId);
+
+		_checkDestinationGroup(group, groupIds, sourceGroup.getGroupId());
+
+		long[] entryIds = ParamUtil.getLongValues(actionRequest, "entryIds");
+
+		for (long entryId : entryIds) {
+			try {
+				DLFileEntry dlFileEntry =
+					_dlFileEntryLocalService.fetchDLFileEntry(entryId);
+
+				if (dlFileEntry != null) {
+					_dlAppService.copyFileEntry(
+						dlFileEntry.getFileEntryId(), destinationFolderId,
+						destinationRepositoryId,
+						dlFileEntry.getFileEntryTypeId(), groupIds,
```

Comment:
Hi, 

There is a reason to change the code from the 

```
_dlAppService.copyFileEntry(
	fileEntryId, destinationFolderId, destinationRepositoryId,
	_getFileEntryTypeId(destinationRepositoryId, fileEntryId),
	groupIds,
	ServiceContextFactory.getInstance(
		DLFileEntry.class.getName(), actionRequest));	
```	
to this?

On this case we do not check the file entry id for the files so it will fail and we should copy with the basic document type

https://github.com/liferay-lima/liferay-portal/assets/47524751/9a254e64-c3e7-4780-95fd-ab695cee0a0b

Reply:
That was a mistake. I'll bring that logic back.